provider/openstack/provider.go

```diff
@@ -168,6 +168,13 @@ func (p EnvironProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+// Ping tests the connection to the cloud, to verify the endpoint is valid.
+func (p EnvironProvider) Ping(endpoint string) error {
+	c := client.NewClient(&identity.Credentials{URL: endpoint}, 0, nil)
```

Comment:
I had to double check, but no, there is no Close method.  Good to check, though, since I hadn't thought of it.